tee-worker/litentry/core/data-providers/src/graphql.rs

```diff
@@ -179,12 +216,56 @@ impl GraphQLClient {
 			Err(Error::GraphQLError("Invalid GraphQL response".to_string()))
 		}
 	}
+
+	pub fn query_total_transactions(
+		&mut self,
+		credentials: VerifiedCredentialsTotalTxs,
+	) -> Result<Vec<TotalTxsStruct>, Error> {
+		let path = "latest/graphql".to_string();
+		let query_value = credentials.to_graphql();
+		let query = vec![("query", query_value.as_str())];
+		let response = self
+			.client
+			.get_with::<String, QLResponse>(path, query.as_slice())
+			.map_err(|e| Error::RequestError(format!("{:?}", e)))?;
+
+		let mut result: HashMap<String, TotalTxsStruct> = HashMap::new();
+
+		response.data.get("data").and_then(|v| v.as_object()).and_then(|map| {
+			for (_network, list) in map {
+				list.as_array().and_then(|element| {
+					for x in element {
+						// aggregate total_transactions from different networks, like group_by.
+						if let Ok(obj) = serde_json::from_value::<TotalTxsStruct>(x.clone()) {
+							if result.contains_key(&obj.address) {
+								result.get_mut(&obj.address).and_then(|origin| {
+									origin.total_transactions += obj.total_transactions;
+									None::<u8>
+								});
+							} else {
+								result.insert(obj.address.clone(), obj.clone());
+							}
+						}
```

Comment:
why returns a `None::<u8>`?
```
if let Some(origin) = result.get_mut(&obj.address) {
    origin.total_transactions += obj.total_transactions;
} else {
    result.insert(obj.address.clone(), obj.clone());
}
```

Reply:
Yes, you are right. I forgot to write code like this. Too many and_then 😅